### PR TITLE
[NEEDS REVIEW] [INSERT NAME HERE] revisited

### PR DESCRIPTION
**Pull request by MadmanMartian**
revisit of #14499

Discussion issue/thread: #10403

Not too sure where to start, so fixed the compile issues and am going to sleep on it

**Status Update**
~~They're functional. They do need some TLC on the HUD, and how they show their weapons, as the arms can sometimes overlay themselves above the held item sprites~~

Lot of sanity apparently required

**TODO**

- [x] It's Life() Jim, but not as we know it
- [x] How the fuck do squid breath?
- [x] How the fuck do squid deal with damage?
- [x] How the fuck do squid do anything, for that matter?
- [x] Slips are the number one enemy to squids, as they have no slot besides a hat slot and hands
- [x] How the fuck do we find the squid?

**FIX LIST**
- [x] Immune to most diseases
- [x] Immune to quite a few wizard spells
- [x] Can't use bodybags
- [x] Immune to mousetraps
- [x] Can't use megaphones
- [x] Immune to slime stunning from melee combat
- [x] Can't put people in sleepers
- [x] Can't use the null rod
- [x] Even if you do give them the ID slot back, it won't function on doors/vending machines while worn
- [x] Can't be buckled into wheelchairs

**IT'S A FEAAATUUUURE**
- Permanently statued if touching a cockatrice. The same can be said for most other animals.
- Immune to facehugging / alien disease. No face to hug
- Immune to genetics. Not the same genetic structure. Whichever deity dropped species with similar genetic code all across the galaxy must have something nefarious in mind
- They don't have blood
- hasorgans() returns false because they aren't human - Because no surgery. Maybe they have no organs?
- No surgery - Because hasorgans returns ishuman
- No defibbing - 'Octopuses have three hearts; a systemic heart that circulates blood round the body and two branchial hearts that pump it through each of the two gills.' - https://en.wikipedia.org/wiki/Octopus
- Can't be wrapped in wrapping paper
- Immune to shard damage - Unable to sort this without making the PR unatomic

### CODE REVIEW

File: vgstation13.dme
Comment (jknpj) [inline vgstation13.dme]: why

File: cfg/admin.txt
Comment (Exxion) [inline cfg/admin.txt]: First of all, fuck this file. It's started appearing recently. I don't really know why. BYOND update or something? It got added to `.gitignore` but apparently everyone still has to do something to refresh their repos or something to make it stop adding itself. I don't really know how Git works, honestly.

Second, this is one of the worst ckeys I've ever seen

**Comment by ghost** [comment]
dear god

File: cfg/admin.txt
Comment (Intigracy) [inline cfg/admin.txt]: Based on the only two names currently in the file, I wonder who caused it.

Comment (Intigracy) [reply cfg/admin.txt]: This file didn't exist until https://github.com/vgstation-coders/vgstation13/commit/211a1db8326bd16bcbbf394f9c4a3f0e3dae1eb3 and actually isn't in .gitignore

Comment (Intigracy) [reply cfg/admin.txt]: Okay, this file specifically is used for access to the Reboot/Profile buttons through DM.

Also it is in .gitignore my bad, people just need to update their .gitignores since pulling from the repo doesn't work since .gitignore ignores itself lmao.

File: vgstation13.dme
Comment (SonixApache) [reply vgstation13.dme]: yeah, don't do this

**Comment by Riekland** [comment]
how long until he kills himself via two shots to the back of his head

File: vgstation13.dme
Comment (Kurfursten) [reply vgstation13.dme]: Obviously the reason it's included is that he was doing testing

It's good to point out so he doesn't forget to remove it, but you've been around to know 'why'

Comment (MadmanMartian) [reply vgstation13.dme]: Sorry, it was in the merge commit along with the admin file. This is all basically raw pulled from unids old PR

**Comment by SonixApache** [comment]
end this fucking name

**Comment by SonixApache** [comment]
thank you

**Comment by ghost** [comment]
Don't you disrespect the dead like that @SonixApache

**Comment by SonixApache** [comment]
your memes are stale and unfunny

**Comment by ghost** [comment]
theyre literally called scootaloos change the title back

File: code/modules/mob/living/carbon/martian/martian.dm
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/martian/martian.dm]: Are people satisfied now?

Comment (SonixApache) [reply code/modules/mob/living/carbon/martian/martian.dm]: no

**Comment by JUSTADDITYOUMEMELORDS** [comment]
JUST ADD IT YOU MEME LORDS

**Comment by MusgardTiger** [comment]
**How the fuck do squid breath?**
They should breathe good ole oxygen but I'll leave how to deal with the unfortunate reality of having to stick an oxygen tank up their throats.

 **How the fuck do squid deal with damage?**
Their limbs should be very strong but their floff bodies should be weaker. They should also have every organ they need in their heads. Hull breaches are also very common so are Martians going to get fucked over whenever there is a hull breach? I guess they can rely on chemicals?


 **How the fuck do squid do anything, for that matter?**
Having a lot of hands means they can carry a lot of containers, so that can be very useful for literally any department other than security and engineering. So if they are gonna be a station race they should be restricted from engineering and security because engineers are dependant on rig suits more or less for anything they do other than antimatter engine and security cannot afford to spill their stuff on the ground whenever they get stunned/disarmed. Their bites should be powerful because of their beaks.

 **Slips are the number one enemy to squids, as they have no slot besides a hat slot and hands**
They need some sort of armband for id cards if they are gonna be station race, people disarm for fun and it'll not be so fun for scondaloos if they drop their id along with everything they carry whenever somebody gets bored.

 **How the fuck do we find the squid?**
I was thinking about an outpost like trader outpost but now that I am thinking, it'd be better if we had on station tbqh mainly because I have no idea for what martians can do in an outpost that current races can't.

**Comment by SonixApache** [comment]
just make them stun proof (as in, can't fall over, but still immobilized)

File: code/modules/mob/living/carbon/martian/martian.dm
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/martian/martian.dm]: This does not work, and I'm not too sure why.

They just do not get the stat tab

File: code/modules/mob/living/carbon/martian/martian.dm
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/martian/martian.dm]: Attempted to use this to try and get the HUD to update appropriately (Possessing a thingum when it was already holding something did not show what it was holding/wearing in the HUD slots) but has not worked

File: code/modules/mob/living/carbon/martian/update_icons.dm
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/martian/update_icons.dm]: Now when update_icon is called, it shows what is being worn in the head on the HUD, but the icon overlay on the character is lost

**Comment by MadmanMartian** [comment]
A binary file conflict? How horrible!

**Comment by Intigracy** [comment]
By making them not a species of human they're going to fail every ishuman() check.
If you intend them to behave like monkies, the same is true for all the ismonkey() checks.

Some of these may be intended, but I doubt all of them are.

* Immune to most diseases
* Immune to facehugging / alien disease
* Permanently statued if touching a cockatrice
* Immune to quite a few wizard spells
* They don't have blood
* Can't use bodybags
* Immune to mousetraps
* Can't use megaphones
* Immune to slime stunning from melee combat
* Can't put people in sleepers
* Can't be wrapped in wrapping paper
* No surgery
* Can't use the null rod
* Even if you do give them the ID slot back, it won't function on doors/vending machines while worn
* hasorgans() returns false because they aren't human
* Can't be buckled into wheelchairs
* Immune to genetics
* Immune to shard damage
* No defibbing

There's a LOT more

**Comment by MadmanMartian** [comment]
Oh, wow, you've got a point there

File: code/modules/mob/living/carbon/martian/inventory.dm
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/martian/inventory.dm:68]: Do doors and vending machines not use this then?

Comment (Intigracy) [reply code/modules/mob/living/carbon/martian/inventory.dm:68]: That one probably works then.

**Comment by Shadowmech88** [comment]
It'd probably be best to make them a species of human and just alter human code so that it doesn't assume stuff like the presence of limbs or what inventory slots a human has. Such things could be moved into species datums.

**Comment by SonixApache** [comment]
In my opinion 80% of those `isHuman()` issues wouldn't be a problem, and the other 20% wouldn't be worse if rewritten to not check for `isHuman()`.

**Comment by Intigracy** [comment]
It'd also eliminate a majority of the copypaste.

**Comment by clusterfack** [comment]
It's funny, you guys are slowly recreating composition instead of inheritance through the use of macros.

**Comment by Exxion** [comment]
Really shitty composition, but I guess

**Comment by clusterfack** [comment]
Scootaloos when

**Comment by Iamgoofball** [comment]
was i pinged at any point here, github is marking this as a participating for whatever reason

File: cfg/admin.txt
Comment (PJB3005) [reply cfg/admin.txt]: > Also it is in .gitignore my bad, people just need to update their .gitignores since pulling from the repo doesn't work since .gitignore ignores itself lmao.

That's only if you gitignore the gitignore, which is already your own fault.

**Comment by MadmanMartian** [comment]
general plan for the last tick is something that requires the infamous antag datums, so stay tuned

**Comment by SonixApache** [comment]
squid datums?

**Comment by PJB3005** [comment]
Splatoon in SS13 when?

**Comment by SonixApache** [comment]
soon as this is merged, i guess

**Comment by PJB3005** [comment]
But where is my ink and ability to be a kid?

**Comment by SonixApache** [comment]
pen + genetics

**Comment by MadmanMartian** [comment]
oh bollocks

**Comment by MadmanMartian** [comment]
I'm not sure how, but I managed to entirely remove my work on martians

**Comment by MadmanMartian** [comment]
and that's fixed it.

Not sure how, but it's fixed

**Comment by SonixApache** [comment]
wait what

**Comment by MadmanMartian** [comment]
Can technically 'find' squid now, as an equivalent to the mommi fab now added

**Review by N3X15 — APPROVED** [review]

**Comment by Shadowmech88** [comment]
Just out of curiosity, why were these made `/carbon`s rather than a species of human?
I think I've asked before, but I don't remember what the answer was.

**Comment by MadmanMartian** [comment]
For them to be made a subtype of human would be to strip out a lot of vars, such as the inventory slots, redo of the hand slots, etc.

Was easier to just make them a subtype of carbon and only add half the fluff, than make them a subtype of human and spend more time filtering the spaghetti code to be pure penne

**Comment by Shadowmech88** [comment]
I imagine a lot of `ishuman()` checks were made into `iscarbon()` in order to allow the martians to do things.
Have the abilities of monkeys, slimes, and xenos, which are also carbons, changed due to this?

**Comment by MadmanMartian** [comment]
nope, ismartian() was added alongside most of the ishuman checks

**Comment by Shadowmech88** [comment]
It might have been a better idea to make an `ishigherbeing()` or something check which calls both `ishuman()` and `ismartian()`.

**Comment by MadmanMartian** [comment]
you've had 2 months to make these sorts of suggestions, why wait until it's merged?

**Comment by Shadowmech88** [comment]
I mostly just forgot it existed until it was merged and everyone started talking about it.
I don't expect you to go back and change it now, I was just offering a thought.

**Pull request merged**
